Symfony/CS/Fixer/Contrib/Php4ConstructorFixer.php

```diff
@@ -0,0 +1,167 @@
+                    // That's pretty bad, delete the new one
+                    for ($i = $p5['start']; $i <= $p5['end']; $i++) {
+                        $tokens[$i]->clear();
+                    }
+                    // And rename the old one
+                    $tokens[$p4['name']]->setContent('__construct');
+                }
+            }
+        }
+
+        // Check calls to the parent constructor
+        foreach ($tokens->findGivenKind(T_EXTENDS) as $index => $token) {
+            $parIndex = $tokens->getNextMeaningfulToken($index);
+            $parentClass = $tokens[$parIndex]->getContent();
+
+            // using parent::ParentClassName()
+            $i = $parIndex;
+            while ($i = $tokens->getNextTokenOfKind($i, array(array(T_PAAMAYIM_NEKUDOTAYIM, "::")))) {
+                $p = $tokens->getPrevMeaningfulToken($i);
+                if ($tokens[$p]->equals(array(T_STRING, 'parent'))) {
+                    $i = $tokens->getNextMeaningfulToken($i);
+                    if (!strcasecmp($tokens[$i]->getContent(), $parentClass)) {
+                        // Found parent
+                        $tokens[$i]->setContent('__construct');
+                        break;
+                    }
+                }
+            }
+
+            // using $this->ParentClassName()
+            $i = $parIndex;
+            while ($i = $tokens->getNextTokenOfKind($i, array(array(T_OBJECT_OPERATOR, "->")))) {
+                $p = $tokens->getPrevMeaningfulToken($i);
+                if ($tokens[$p]->equals(array(T_VARIABLE, '$this'))) {
+                    $op = $i;
+                    $i = $tokens->getNextMeaningfulToken($i);
+                    if (!strcasecmp($tokens[$i]->getContent(), $parentClass)) {
+                        // Found parent
+                        $tokens[$p]->setContent('parent');
+                        $tokens[$op]->setContent('::');
+                        $tokens[$i]->setContent('__construct');
+
+                        break;
+                    }
+                }
+            }
+        }
+
+        return $tokens->generateCode();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return 'php4_constructor';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return 'Convert PHP4-style constructors to __construct.';
+    }
+
+    /**
+     * Find a function matching a given name within certain bounds.
+     *
+     * @param  Tokens      $tokens
+     * @param $name
+     * @param $start
+     * @param $end
+     * @return array|false
+     */
+    protected function findFunction(Tokens $tokens, $name, $start, $end)
+    {
+        foreach ($tokens->findGivenKind(T_FUNCTION) as $index => $token) {
+            if ($index < $start || $index > $end) {
+                continue;
+            }
+
+            $funcIndex =  $tokens->getNonWhitespaceSibling($index, 1);
```

Comment:
- please use shorthand `getNext...`
- it will break if the next token is comment. please add that case to tests and then fix it